JSR_Shopping_List_Functions-master/js/app.js

```diff
@@ -0,0 +1,947 @@
+// 				'item': 'test',
+// 				'price': 1
+// 			});
+// 			list = addToShoppingList({
+// 				'item': 'test2',
+// 				'price': 2
+// 			}, list);
+// 			list = removeFirstItem(list);
+
+// 			chai.assert.equal(list.length, 1)
+// 			// asert only item in list is 'test2' with price 2
+// 			chai.assert.equal(list[0].item, 'test2')
+// 			chai.assert.equal(list[0].price, 2)
+// 		})
+// 	});
+
+// 	/* 5
+// 		@function removeNthItem
+// 		@param i {number}
+// 		@param list {array, []}
+// 		@returns list
+// 		@description
+// 			given `i`, an index < length of list
+// 			remove that item from list and return the
+// 			resulting list
+// 			if...
+// 				- what if `i` > length of list
+// 				- `i` is < 0
+// 				- `i` is not number
+// 			^^ return error
+// 	*/
+
+// 	// implement function here
+// 	const removeNthItem = (i, list) => {
+// 		if (i < 0|| typeof i === 'string' || i > list.length) {
+// 			console.log('lessthan0');
+// 			throw new Error ('needs to be a number and in the array');
+// 		}
+// 		else{
+// 			list.splice(i, 1);
+// 			return list;	
+// 		}
+// 	}
+
+
+	
+// 	// TEST
+// 	describe('5. removeNthItem', () => {
+// 		it('should remove i-th item from list', () => {
+// 			let list = addToShoppingList({
+// 				'item': 'test',
+// 				'price': 1
+// 			});
+// 			list = addToShoppingList({
+// 				'item': 'test2',
+// 				'price': 2
+// 			}, list);
+// 			list = addToShoppingList({
+// 				'item': 'test3',
+// 				'price': 3
+// 			}, list);
+
+
+// 			list = removeNthItem(1, list);
+
+// 			chai.assert.equal(list.length, 2)
+
+// 			chai.assert.equal(list[0].item, 'test')
+// 			chai.assert.equal(list[0].price, 1)
+
+// 			chai.assert.equal(list[1].item, 'test3')
+// 			chai.assert.equal(list[1].price, 3)
+// 		});
+
+// 		it('should throw error if i < 0', () => {
+// 			// if i < 0
+// 			chai.assert.throws(() => {
+// 				removeNthItem(-1, [])
+// 			}, Error);
+// 		});
+
+// 		it('should throw error if i > length of list', () => {
+// 			// if i > length of array
+// 			chai.assert.throws(() => {
+// 				removeNthItem(1, [])
+// 			}, Error);
+// 		});
+
+// 		it('should throw error if i is not a number', () => {
+
+// 			// if i is not a number
+// 			chai.assert.throws(() => {
+// 				removeNthItem('adfas', [])
+// 			}, Error);
+// 		})
+// 	});
+
+	/* 6
+		@function removeNItems
+		@param i {number}
+		@param num {number}
+		@param list {array, []}
+		@returns list
+		@description
+			same as above but now we wish to remove ALL
+			items from i to i+num and return the resulting list
+			if...
+				- `i` < 0
+				- `i` or `num` is not a number
+				- `i+num` > length of list
+				- `num` > length of list
+			^^ return error
+	*/
+
+	// // implement function here
+	// 	const removeNItems = (i, num, list) =>{
+	// 			if(i < 0 || typeof (i + num) === 'string' || num > list.length || (i+num) > list.length ){
+	// 			throw new error('bad');
+	// 			}
+	// 			else{
+
+	// 			list.splice(i, num);
```

Comment:
Now this one is a little tricky. The correct answer here is `list.splice(i, i+num);` because we want `i` to `num`, _inclusive_. If we look at the description on [Line 270](https://github.com/FEWDMaterials/JSR_Shopping_List_Functions/pull/14/files#diff-f23f1b7e90753c7d2e5ede665f475688R270), we see that it's spelling that out.